src/parsing.jl

```diff
@@ -316,14 +316,66 @@ function pqparse(::Type{InfExtendedTime{T}}, str::AbstractString) where T<:Dates
 end
 
 # UNIX timestamps
-function Base.parse(::Type{DateTime}, pqv::PQTextValue{PQ_SYSTEM_TYPES[:int8]})
+function Base.parse(::Type{DateTime}, pqv::PQValue{PQ_SYSTEM_TYPES[:int8]})
     return unix2datetime(parse(Int64, pqv))
 end
 
-function Base.parse(::Type{ZonedDateTime}, pqv::PQTextValue{PQ_SYSTEM_TYPES[:int8]})
+function Base.parse(::Type{ZonedDateTime}, pqv::PQValue{PQ_SYSTEM_TYPES[:int8]})
     return TimeZones.unix2zdt(parse(Int64, pqv))
 end
 
+# All postgresql timestamptz are stored in UTC time with the epoch of 2000-01-01.
+const POSTGRES_EPOCH_DATE = Date("2000-01-01")
+const POSTGRES_EPOCH_DATETIME = DateTime("2000-01-01")
+
+# Note: Because postgresql stores the values as a Microsecond in Int64, the max (infinite)
+# value of date time in postgresql when querying binary is 294277-01-09T04:00:54.775
+# and the minimum is -290278-12-22T19:59:05.225.
```

Comment:
Does this mean that PostgreSQL is returning its maximums instead of infinity? 

We may want to translate its maximums to our maximums so we're consistent about handling infinity. 

Any relevant documentation about this?

Reply:
The code is a bit hard to follow, but I think with the comment and code here is makes it clear that PostgreSQL max/min timestamp is based on int64. 

https://github.com/postgres/postgres/blob/c30f54ad732ca5c8762bb68bbe0f51de9137dd72/src/include/datatype/timestamp.h#L106-L110

So to be clear, you want me to make the change [here](https://github.com/invenia/LibPQ.jl/blob/master/src/parsing.jl#L244) to match what postgres max would be?

```julia
const POSTGRES_EPOCH_DATETIME = DateTime("2000-01-01")
const POSTGRES_DATETIME_MAX = POSTGRES_EPOCH_DATETIME + typemax(Microsecond)
```

Reply:
Playing around with timestamps in postgres, I found that the max is `294276-12-31T23:59:59.999`

```
eis_data=> SELECT timestamp with time zone '294277-01-01T00:00:00.000';
ERROR:  timestamp out of range: "294277-01-01T00:00:00.000"
LINE 1: SELECT timestamp with time zone '294277-01-01T00:00:00.000';
                                        ^
eis_data=> SELECT timestamp with time zone '294276-12-31T23:59:59.999';
         timestamptz          
------------------------------
 294276-12-31 23:59:59.999+00
(1 row)
```